tests/ocpp_tests/test_sets/ocpp16/ocpp_compliance_tests.py

```diff
@@ -6589,3 +6595,973 @@ async def test_signed_update_firmware(
         "SignedFirmwareStatusNotification",
         call.SignedFirmwareStatusNotificationPayload(FirmwareStatus.installed, 1),
     )
+
+
+@pytest.fixture
+def probe_module(started_test_controller, everest_core) -> ProbeModule:
+    # initiate the probe module, connecting to the same runtime session the test controller started
+    module = ProbeModule(everest_core.get_runtime_session())
+
+    return module
+
+
+@pytest_asyncio.fixture
+async def chargepoint_with_pm(central_system: CentralSystem, probe_module: ProbeModule):
+    """Fixture for ChargePoint16. Requires central_system_v201 and test_controller. Starts test_controller immediately
+    """
+    # wait for libocpp to go online
+    cp = await central_system.wait_for_chargepoint()
+    yield cp
+    await cp.stop()
+
+
+class ProbeModuleCostAndPriceMetervaluesConfigurationAdjustment(EverestConfigAdjustmentStrategy):
+    """
+    Probe module to be able to 'inject' metervalues
+    """
+    def adjust_everest_configuration(self, everest_config: Dict):
+        adjusted_config = deepcopy(everest_config)
+
+        adjusted_config["active_modules"]["grid_connection_point"]["connections"]["powermeter"] = [
+            {"module_id": "probe", "implementation_id": "ProbeModulePowerMeter"}]
+        adjusted_config["active_modules"]["evse_manager"]["connections"]["powermeter_grid_side"] = [
+            {"module_id": "probe", "implementation_id": "ProbeModulePowerMeter"}]
+
+        return adjusted_config
+
+
+class ProbeModuleCostAndPriceDisplayMessageConfigurationAdjustment(EverestConfigAdjustmentStrategy):
+    """
+    Probe module to be able to mock display messages
+    """
+    def adjust_everest_configuration(self, everest_config: Dict):
+        adjusted_config = deepcopy(everest_config)
+
+        adjusted_config["active_modules"]["ocpp"]["connections"]["display_message"] = [
+            {"module_id": "probe", "implementation_id": "ProbeModuleDisplayMessage"}]
+
+        return adjusted_config
+
+
+class ProbeModuleCostAndPriceSessionCostConfigurationAdjustment(EverestConfigAdjustmentStrategy):
+    """
+    Probe module to be able to mock the session cost interface calls
+    """
+    def adjust_everest_configuration(self, everest_config: Dict):
+        adjusted_config = deepcopy(everest_config)
+
+        adjusted_config["active_modules"]["probe"]["connections"]["session_cost"] = [
+            {"module_id": "ocpp", "implementation_id": "session_cost"}]
+
+        return adjusted_config
```

Comment:
These seem to be defined similar for ocpp16 and ocpp201 so I think they can be moved to some common file and the be imported